### PR TITLE
ci: we want PR tested in travis ci

### PR DESCRIPTION
**Pull request by thalman**
Currently there is no automated CI for authselect. The .travis.yml file shows how to do at least
'make check' and 'make distcheck' on PR automatically.

Resolves:
https://github.com/pbrezina/authselect/issues/5

### CODE REVIEW

**Comment by thalman** [comment]
@pbrezina, please review. I created PR in my repo (my branch -> my master) and tested integration.
Here you can see it: https://travis-ci.com/thalman/authselect/jobs/182654143

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+sudo: false
+
+services:
+- docker
+
+before_install:
+- docker pull fedora
+- docker run -d --rm --name AUTHSELECT-FEDORA -v `pwd`:/buildroot fedora /bin/sh -c "tail -f /dev/null"
+- docker ps
+- docker exec AUTHSELECT-FEDORA dnf -y install gcc make gettext gettext-devel autoconf automake libtool m4 pkgconfig popt-devel asciidoc libcmocka-devel libselinux-devel file
```
Comment (pbrezina) [inline .travis.yml]: Is the `file` at the end of the line a typo or is it actually doing something?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+script:
+- docker exec AUTHSELECT-FEDORA /bin/sh -c "cd /buildroot && autoreconf -iv && ./configure --enable-silent-rules"
+- docker exec AUTHSELECT-FEDORA /bin/sh -c "cd /buildroot && make"
+- docker exec AUTHSELECT-FEDORA /bin/sh -c "cd /buildroot && make check"
+- docker exec AUTHSELECT-FEDORA /bin/sh -c "cd /buildroot && make distcheck"
```
Comment (pbrezina) [inline .travis.yml]: Perhaps adding `make rpms` also makes sense?

**Comment by pbrezina** [comment]
Thank you Tomas, this is really helpful.

File: .travis.yml (hunk 1 above)
Comment (thalman) [reply .travis.yml]: It is not typo, file utility is used somehow in configure. Configure complained when running (some pipe check) that file utility is not present.

File: .travis.yml (hunk 2 above)
Comment (thalman) [reply .travis.yml]: > Perhaps adding `make rpms` also makes sense?

Done

**Comment by thalman** [comment]
@pbrezina I added some packages for rpm build + `make rpms`

**Comment by pbrezina** [comment]
Thank you.

Pushed to master:
* 6275dcf3f6b3f134b14d6b1e27d5cc6a69edf3cb

**Comment by pbrezina** [comment]
How do I enable it?

**Comment by thalman** [comment]
I used https://github.com/mbonaci/mbo-storm/wiki/Integrate-Travis-CI-with-your-GitHub-repo#integrate-your-github-project-with-travis-ci
It is a bit outdated but +- OK

**Pull request closed without merging**